src/ds5/ds5-factory.cpp

```diff
@@ -167,6 +187,60 @@ namespace librealsense
         };
     };
 
+    class rs416_device : public ds5_rolling_shutter,
+        public ds5_active, public ds5_advanced_mode_base
+    {
+    public:
+        rs416_device(std::shared_ptr<context> ctx,
+            const platform::backend_device_group& group,
+            bool register_device_notifications)
+            : device(ctx, group, register_device_notifications),
+            ds5_device(ctx, group),
+            ds5_rolling_shutter(ctx, group),
+            ds5_active(ctx, group),
+            ds5_advanced_mode_base(ds5_device::_hw_monitor, get_depth_sensor()) {}
+
+        std::shared_ptr<matcher> create_matcher(const frame_holder& frame) const override;
+
+        std::vector<tagged_profile> get_profiles_tags() const override
+        {
+            std::vector<tagged_profile> tags;
+            auto usb_spec = get_usb_spec();
+            if (usb_spec >= platform::usb3_type || usb_spec == platform::usb_undefined)
+            {
+                tags.push_back({ RS2_STREAM_DEPTH, -1, 720, 720, RS2_FORMAT_Z16, 30, profile_tag::PROFILE_TAG_SUPERSET | profile_tag::PROFILE_TAG_DEFAULT });
+                tags.push_back({ RS2_STREAM_INFRARED, 1, 1024, 1024, RS2_FORMAT_RAW10, 30, profile_tag::PROFILE_TAG_SUPERSET | profile_tag::PROFILE_TAG_DEFAULT });
+                tags.push_back({ RS2_STREAM_INFRARED, 2, 1024, 1024, RS2_FORMAT_RAW10, 30, profile_tag::PROFILE_TAG_SUPERSET });
+            }
+            else
+            {
+                tags.push_back({ RS2_STREAM_DEPTH, -1, 576, 576, RS2_FORMAT_Z16, 15, profile_tag::PROFILE_TAG_SUPERSET | profile_tag::PROFILE_TAG_DEFAULT });
+                tags.push_back({ RS2_STREAM_INFRARED, 1, 576, 576, RS2_FORMAT_RAW10, 15, profile_tag::PROFILE_TAG_SUPERSET | profile_tag::PROFILE_TAG_DEFAULT });
+                tags.push_back({ RS2_STREAM_INFRARED, 2, 576, 576, RS2_FORMAT_RAW10, 15, profile_tag::PROFILE_TAG_SUPERSET });
+            }
+            return tags;
+        };
+
+        bool contradicts(stream_profile_interface* a, const std::vector<stream_profile>& others) override
+        {
+            if (auto vid_a = dynamic_cast<video_stream_profile_interface*>(a))
+            {
+                for (auto request : others)
```

Comment:
PLs check if those two can be merged as in comment above

Reply:
its probably not required anymore, the only limitation now is matching FPS, will remove